src/watchmaker/workers/yum.py

```diff
@@ -54,40 +45,25 @@ def _get_amazon_el_version(self, version):
 
     def get_dist_info(self):
         """Validate the Linux distro and return info about the distribution."""
-        dist = None
-        version = None
-        el_version = None
-
-        # Read first line from /etc/system-release
+        # Error if 'dist' is not found in SUPPORTED_DISTS
         try:
-            with io.open('/etc/system-release', encoding='utf8') as fh_:
-                release = fh_.readline().strip()
-        except Exception:
-            self.log.critical(
-                'Failed to read /etc/system-release. Cannot determine system '
-                'distribution!'
-            )
-            raise
-
-        # Search the release file for a match against _supported_dists
-        matched = self.DIST_PATTERN.search(release.lower())
-        if matched is None:
+            dist = self.SUPPORTED_DISTS[distro.id()]
+        except KeyError:
             # Release not supported, exit with error
             msg = (
                 'Unsupported OS distribution. OS must be one of: {0}'
-                .format(', '.join(self.SUPPORTED_DISTS))
+                .format(', '.join(self.SUPPORTED_DISTS.keys()))
             )
             self.log.critical(msg)
             raise WatchmakerError(msg)
-
-        # Assign dist,version from the match groups tuple, removing any spaces
-        dist, version = (x.replace(' ', '') for x in matched.groups())
+        version = distro.version()[0]
+        el_version = None
```

Comment:
Once this assignment is again handled at the top of the function, this line can be removed.

Reply:
"This line"?

Reply:
Comments code code in github pull requests are on lines. So yes, "this line" is the line this comment is attached to. It's difficult to refer to specific line numbers, since as changes are made, the line number will change. But at the moment, it is line 60.

Reply:
Become moot, since the refactor to use a function.